json/json.go

```diff
@@ -287,6 +277,19 @@ func (dec *Decoder) readValue() (v []byte, err error) {
 				dec.remain = skipSpaces(r)
```

Comment:
Not relevant to this PR but do we need to update dec.err here similar to https://github.com/segmentio/encoding/pull/19/files#diff-0839e8ccea528964240d6791f5a24cafR312?

Reply:
No, `dec.err` is supposed to retain the last non-nil error, if we update it here it'll erase the last error seen by the decoder (since we just successfully parsed a JSON value).